index.rst

```diff
@@ -1,26 +1,363 @@
+away from the service-specific availability endpoint, and
+to the corresponding endpoint on the central availability service.
 
-  Sections
-  ========
+The central availability service could be directly exposed externally,
+or only through the redirected external per-service endpoints; this is TBD.
+An advantage to exposing it directly is that it provides a natural home for
+delivering status information for non-IVOA services (e.g., Nublado) that are
+not constrained to have their endpoints registered in VOSI-like ways.
+Any "internal" service-specific availability endpoints, provided by the specific
+IVOA service pods themselves, would not be exposed externally at all.
 
-  and
+For services for which throttling / rate-limiting are enabled, if that is
+implemented in a generic way across the RSP's web services, the central
+availability service should be able to be extended to return a message
+concerning whether rate-limiting is currently being applied.
 
-  Subsections
-  -----------
+Finally, with a uniform availability service in place, it is easy to imagine
+writing a lightweight "dashboard" / "Christmas tree" status display for all
+the components of the RSP.
 
-  and
+The proposal includes the following specific elements:
 
-  Subsubsections
-  ^^^^^^^^^^^^^^
+Basic /availability XML retrieval from Kubernetes controller
+------------------------------------------------------------
+
+The core element of the proposal is a service which can be configured with
+a list of services to represent, and the logic necessary to query the
+Kubernetes controller for the present state of a service and translate that
+into the ``available`` and ``upSince`` attributes of the VOSI-Availability
+data model.
+
+If a centralized repository of information about planned outages is developed
+in order to support messaging through the new RSP home page and framing,
+this could also be queried in order to support delivery of information about
+planned outages via the ``downAt`` and ``backAt`` attributes for scheduled
+downtimes.
+
+An instance of the ``note`` attribute should be generated representing the
+basic assessment that the service appears to be running.
+If a planned outage is known to the system, a human-readable message about
+the outage, e.g., "planned maintenance 2024-04-01 14:00Z through 2024-04-02 16:00Z",
+or "down for rollout of DR3 2026-10-09" could be included in the ``note``.
+
+The service would need to be able to format this information in the VOSI-prescribed
+(very simple) XML format.
+
+Since the author is aware of the existence of anti-XML attitudes in the RSP
+developer community, we note that the central availability service could be
+designed to respond both in JSON and XML.
+One could imagine 
+``https://(address)/api/central-availability/(service-name)``
+responding in JSON and
+``https://(address)/api/central-availability/(service-name)/xml``
+in XML, with the per-service availability endpoints redirected to the latter.
```

Comment:
The normal way to do this is with an `Accept` header in the request that specifies either `application/json` or `application/xml`.

Reply:
Presumably this would still permit XML to be the default if no `Accept:` header is provided?  I don't _think_ that the IVOA standards require the provision of `Accept:` in order to get conformant responses.

IVOA data-retrieval services (which `/availability` isn't, really) use `&RESPONSEFORMAT=(mime-type-or-other-name)` to ask for specific formats in responses, but that's not an established option for VOSI-availability, so I like your suggestion and I'll reword the proposal around that.

Reply:
Edited accordingly.